velocity_modelling/scripts/plot_velocity_model.py

```diff
@@ -0,0 +1,224 @@
+from enum import Enum
+from pathlib import Path
+from typing import Annotated, Optional
+
+import numpy as np
+import pandas as pd
+import typer
+import yaml
+
+from pygmt_helper import plotting
+from qcore import grid
+from velocity_modelling import bounding_box
+
+app = typer.Typer()
+
+
+class VelocityModelComponent(str, Enum):
+    p_wave = "p_wave"
+    s_wave = "s_wave"
+    density = "rho"
+
+
+@app.command()
+def plot_velocity_model(
+    output_plot_path: Annotated[
+        Path,
+        typer.Argument(
+            help="Path to output velocity model plot.", writable=True, dir_okay=False
+        ),
+    ],
+    vm_params_ffp: Annotated[
+        Optional[Path], typer.Option(help="Path to VM params YAML", dir_okay=False)
+    ] = None,
+    velocity_model_ffp: Annotated[
+        Optional[Path],
+        typer.Option(
+            help="Path to velocity model directory", file_okay=False, exists=True
+        ),
+    ] = None,
+    resolution: Annotated[
```

Comment:
And if the velocity model ffp is no longer optional this should have a default as a number, can simplify some checking too if this is done

Reply:
Even if the velocity model filepath is not None, this should still be. Defaulting to a value like 0.4 or 0.1 means that you will just get an ugly stacktrace when the plotting crashes, instead of a simple sentence that says you forgot to supply the resolution. I could (and originally did) have the resolution sourced only from the vm params yaml or the realisation file; it wasn't a CLI option. However, this meant that if we are in a situation like the 200m Cybershake situation, where we have a VM but no vm params, this script won't be helpful.